gradio/components/image_editor.py

```diff
@@ -320,7 +320,7 @@ def resolve_path(file_or_url: str | None) -> str | None:
             # If an externally hosted image or a URL, don't convert to absolute path
             if self.proxy_url or client_utils.is_http_url_like(input_data):
                 return input_data
-            return str(utils.abspath(input_data))
+            return self.move_resource_to_block_cache(input_data)
```

Comment:
This isn't very intuitive for custom component authors. Is there a better abstraction or perhaps just an alternative name that doesn't reference implementation details?